README.md

```diff
@@ -381,464 +607,224 @@ class MyBaseClass extends WidgetBase<WidgetProperties> {
 
-const MyViewWidgetContainer = Container(MyViewWidget, 'state', { getProperties });
 ```
-
-**Note:** that both the `getProperties` and `getChildren` functions do not need to be provided, if the functions are not defined the default mappers will be used that return an empty object and an empty array respectively.
-
-There may be times when the default `BaseInjector` doesn't fully meet your needs. For example if the context contains a reference to an eventable instance, you may want to add an event listener in the `Injector` to perform some logic, perhaps invalidate the widget.
-
-To do this the `BaseInjector` can be extended easily to add the extra logic required.
-
-```ts
-interface MyContext {
-	eventeableInstance: Evented;
-	bar: number;
-}
-
-class MyInjector extends BaseInjector<MyContext> {
-	constructor(context: MyContext) {
-		super(context);
-		const { eventeableInstance } = context;
-		eventeableInstance.on('change', () => {
-			this.invalidate();
-		});
-	}
-
-	protected toInject(): any {
-		const { eventeableInstance, bar } = this.context;
-		return {
-			eventeableInstance,
-			bar
-		};
-	}
-}
+const hasRootBeenRendered = this.meta(Dimensions).has('root');
 ```
 
-#### Theming
-
-##### Overview
-
-Widgets are themed using `css-modules` and the `Themeable` mixin. Each widget must implement a .css file that contains all the css classes that will be used to style it. The `baseClasses` object is the css API for the Widget: `baseClasses` css classes can be overridden by external themes. Further customization of specific Custom Widget classes can be achieved by passing `extraClasses` into the widget.
-The `Themeable` mixin provides a `classes` function that controls the classes to be applied to each node. Classes from the base `css` object passed to the `classes` function can be themed and overridden. To create fixed classes that cannot be changed, the chained `fixed` function can be used.
-
-##### Authoring a Base Theme
-
-A base theme is authored using `css-modules` and `cssnext`. The base theme `css` file should be located in a `styles` folder within the Widget's package directory.
-The `typed-css-modules` [cli](https://github.com/Quramy/typed-css-modules#cli) should be used in `watch` mode in order to generate typings for TypeScript usage. This is automatically included within the `dojo build -w` command from `dojo-cli`.
-
-```
-tabPanel
-├── createTabPanel.ts
-└── styles
-    └── tabPanel.css
-```
+##### Implementing Custom Meta
 
-The `baseClasses` css must contain a complete set of all of the classes you wish to apply to a widget as all theme and extra classes are limited by the classnames made available here.
-Classnames are locally scoped as part of building a Dojo application. A theme `key` is generated at build time to locate the themes for each class where a theme is set.
+You can create your own meta if you need access to DOM nodes.
```

Comment:
"you need access to DOM nodes" -> perhaps "if you want to preserve reactive principles with a DOM node"?